cmd/tailscale/cli/up.go

```diff
@@ -294,6 +296,29 @@ func runUp(ctx context.Context, args []string) error {
 		},
 	}
 
+	if upArgs.json {
+		opts.Notify = func(n ipn.Notify) {
+			if n.ErrMessage != nil {
+				fmt.Fprintf(os.Stdout, `{"error":%q}`, *n.ErrMessage)
+				os.Exit(1)
+			}
+			state := ""
+			if n.State != nil {
+				state = n.State.String()
+			}
+			if url := n.BrowseToURL; url != nil {
+				fmt.Fprintf(os.Stdout, `{"state":%q,"auth_url":%q}`, state, *url)
+				cancel()
+				return
+			}
+			if n.State != nil && *n.State == ipn.Running {
+				fmt.Fprintf(os.Stdout, `{"state":%q}`, state)
+				cancel()
```

Comment:
what if it's stopped ("disconnected") but not running and thus there's no authURL? won't this just hang forever with no JSON output?